### PR TITLE
Update GitHub templates

### PR DESCRIPTION
**Pull request by tunetheweb**
This makes the following changes to the New Issue and Pull Request templates:
- Removes the reference to adding labels. Non-maintainers cannot add labels - it's [a long standing issue](https://github.com/isaacs/github/issues/875) apparently - so it's confusing for them, and maintainers should know to add labels.
- Adds a default Pull Request Template. We had 4 templates but annoyingly they are [only used if you add a query param to your compare URL when opening the PR](https://stackoverflow.com/a/62613198/2144578)! I mean who does that?!? So better to have a generic, default one that will be used than four specific ones that won't be used.
- Removes the Meta data from the 4 Pull Request templates because, unlike the Issue templates, [this are not actually used by GitHub for PR templates](https://github.com/isaacs/github/issues/1252#issuecomment-444940753). Which is disappointing as means we can't auto add labels.
- Removes references to CHANGELOG since that should no longer be manually updated now, except for on releases.

Definitely feels like PR Templates have only been half thought through by GitHub at this time!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (barrywhart) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Nitpick: I have sort of a "process" question. We recently added a similar PR template at work with checkbox items. It looked good to me, but one person pointed out that in the GitHub page that lists all open PRs, for each one, it shows how many checkboxes there are and how many are checked. Most PRs won't have all the boxes checked (like if that step is not applicable for their PR).

TBH, I was okay with this, but as a compromise, we ended up removing some of the Markdown checkboxes and instead asking people to copy/paste a ✅ character next to the things they'd done.

Just wanted your thoughts on this, definitely not a blocker.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (barrywhart) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Should we remind people (or can we even provide the starter text for) the "Fixes #XXX" message? Those are really handy for whoever reviews and merges the PR because it (a) provides context for why the PR exists and (b) merging the PR also closes the related issue.

**Review by barrywhart — COMMENTED** [review]
Looks great overall! A couple of small/nitpick questions.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: Yeah it will say 1 out of 4 items. Or something like this. I can see how that might be confusing.

To be honest I'd be OK just removing the tick boxes and have justing this as a list to remind people, rather than something they have to explicitly tick off. I only included them as they were already in the existing - mostly used - templates. It's up to the reviewer to check they've done the appropriate steps (coupled with the automatic checks like no drops in cover), so not like the tick box does anything.

WDYT?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Good idea will add. I take it for granted people do that but not everyone does.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md]: Changed it to a single checkbox so there is still an action on the user to explicitly state they've read the T&Cs :-)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Added.

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md
Comment (barrywhart) [inline .github/PULL_REQUEST_TEMPLATE/bug_fix.md:1]: Originally, I think you were going to delete these existing, specialized PR templates. Did you learn something that changed your mind?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Good spot. Will do.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Yup.

File: .github/PULL_REQUEST_TEMPLATE/release.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE/release.md:3]: This is the release Pull Request. It should be used by maintainers for raising the PR to update config.ini. See the changes to the CONTRIBUTING.md in this PR:

image

What do you reckon? Will maintainers use this? Or just delete (as per your next comment).

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:1]: I guess I'm just hopeful GitHub will eventually support this. But you're right, probably should delete in meantime as they will just get out of date.

Though see comment about the release.md one. Think that would be used? Or just get rid?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE/release.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE/release.md]: OK I've removed this. It's a bit convoluted and realistically any one know knows to use it will now the steps.

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md
Comment (tunetheweb) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:1]: Thought is over and they are gone.

Comment (barrywhart) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:1]: No strong opinion either way about the PR templates, just curious. Reviewing this again now...

File: CONTRIBUTING.md
Comment (barrywhart) [inline CONTRIBUTING.md:150]: Oh, I misinterpreted the mention of CHANGELOG.md earlier. I thought it was the source, but it's the destination.
```suggestion
Copy the text from the draft release into a new, version-numbered section of the [CHANGELOG.md](CHANGELOG.md) file and update the
```

**Review by barrywhart — COMMENTED** [review]
Suggested wording change to clarify handling of the change log.

**Review by barrywhart — APPROVED** [review]
Thanks for your great work improving the contribution and release processes!!

**Comment by barrywhart** [comment]
Updated PR from `main` branch. Will merge once the build passes. (Not that it would fail, but this is enforced by our GitHub policies.)

**Pull request merged**
